tests/unit-tests/gl/test_tessellation_helpers.cpp

```diff
@@ -0,0 +1,131 @@
+ */
+
+#include <glm/gtc/matrix_transform.hpp>
+#include <mir/gl/tessellation_helpers.h>
+#include <mir/test/doubles/mock_renderable.h>
+
+using namespace testing;
+
+namespace geom = mir::geometry;
+namespace mt = mir::test;
+namespace mtd = mir::test::doubles;
+namespace mgl = mir::gl;
+
+class Tessellation : public testing::Test
+{
+public:
+    void SetUp() override
+    {
+        ON_CALL(renderable, screen_position())
+            .WillByDefault(Return(rect));
+    }
+
+    struct BoundingBox
+    {
+        float left, right, top, bottom;
+    };
+
+    static auto bounding_box(mgl::Primitive const& primitive) -> BoundingBox
+    {
+        BoundingBox box{
+            std::numeric_limits<float>::max(),
+            std::numeric_limits<float>::min(),
+            std::numeric_limits<float>::max(),
+            std::numeric_limits<float>::min()};
+
+        for (int i = 0; i < primitive.nvertices; i++)
+        {
+            box.left    = std::min(box.left,    primitive.vertices[i].position[0]);
+            box.right   = std::max(box.right,   primitive.vertices[i].position[0]);
+            box.top     = std::min(box.top,     primitive.vertices[i].position[1]);
+            box.bottom  = std::max(box.bottom,  primitive.vertices[i].position[1]);
+        }
+
+        return box;
+    }
+
+    static void expect_tex_coords_1_or_0(mgl::Primitive const& primitive)
+    {
+        for (int i = 0; i < primitive.nvertices; i++)
+        {
+            for (int axis = 0; axis < 2; axis++)
+            {
+                float const tex_coord = primitive.vertices[i].texcoord[axis];
+                EXPECT_THAT(tex_coord == 0.0 || tex_coord == 1.0, Eq(true));
+            }
+        }
+    }
+
+    geom::Rectangle const rect{{4, 6}, {10, 20}};
+    NiceMock<mtd::MockRenderable> const renderable;
+};
+
+TEST_F(Tessellation, has_4_vertices)
+{
+    mgl::Primitive const primitive = mgl::tessellate_renderable_into_rectangle(renderable, {});
+    EXPECT_THAT(primitive.type, Eq(GL_TRIANGLE_STRIP));
+    EXPECT_THAT(primitive.nvertices, Eq(4));
+}
+
+TEST_F(Tessellation, has_correct_bounding_box)
+{
+    mgl::Primitive const primitive = mgl::tessellate_renderable_into_rectangle(renderable, {});
+    auto const box = bounding_box(primitive);
+    EXPECT_THAT(box.left, Eq(rect.top_left.x.as_int()));
+    EXPECT_THAT(box.top, Eq(rect.top_left.y.as_int()));
+    EXPECT_THAT(box.right, Eq(rect.right().as_int()));
+    EXPECT_THAT(box.bottom, Eq(rect.bottom().as_int()));
+}
+
+TEST_F(Tessellation, bounding_box_gets_offset)
+{
+    int const x{3}, y{8};
+    mgl::Primitive const primitive = mgl::tessellate_renderable_into_rectangle(renderable, {x, y});
+    auto const box = bounding_box(primitive);
+    EXPECT_THAT(box.left, Eq(rect.top_left.x.as_int() - x));
+    EXPECT_THAT(box.top, Eq(rect.top_left.y.as_int() - y));
+    EXPECT_THAT(box.right, Eq(rect.right().as_int() - x));
+    EXPECT_THAT(box.bottom, Eq(rect.bottom().as_int() - y));
```

Comment:
I guess this is OK, but the test would likely read better with a custom matcher for `BoundingBox`. Vis:
```c++
EXPECT_THAT(bounding_box(primitive), BoundingBoxEq(rect.top_left.x.as_int() - x, rect.top_left.y.as_int() - y, rect.right().as_int() - x, rect.bottom().as_int() - y));